### PR TITLE
Self types appear as inherited in docs and Docs leaking self type methods

### LINKED ISSUE #13203: Docs leaking self type methods
## Compiler version

3.0.1

## Docs
In https://dotty.epfl.ch/api/scala/quoted/Quotes.html we see the methods `unpickleExpr`, `unpickleType`, `ExprMatch` and `TypeMatch` that are part of the self type but not inherited. As such users do not have access to these methods nor should know about them. 

Screenshot 2021-07-29 at 14 17 42

Screenshot 2021-07-29 at 14 21 52

Implementors should know about those methods because they know about the self type.

## Expectation

The method from the self type should not be shown.

### LINKED ISSUE #13204: Self types appear as inherited in docs
## Compiler version

3.0.1

## Minimized code

In the documentation of `scala.quoted.Quotes` (https://dotty.epfl.ch/api/scala/quoted/Quotes.html) we see the self types as if they are inherited.

```Scala
trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
   ...
}
```
Screenshot 2021-07-29 at 14 26 50

## Expectation

Sefl types should not appear as inherited as those are not yet inherited. They guarantee that the implementation will inherit them.

We should show the self type in some form. Maybe the first code snippet could be charged for classes that have an explicit self type
```diff
- trait Quotes
+ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching }
```
Or maybe put it in its own section as most users will not care about this information.

### PR DESCRIPTION
**Pull request by KacperFKorban**
fixes #13204
fixes #13203

### CODE REVIEW

**Comment by KacperFKorban** [comment]
Added a new tab next to `Supertypes`. `Quotes` looks like this now:
image

**Comment by nicolasstucki** [comment]
Looks good.

Minor suggestion: the `self` in `self: QuoteUnpickler & QuoteMatching` does not convey any useful information to the user. A user that will extend this class can rename the self type if he wishes. Maybe we should just display `QuoteUnpickler & QuoteMatching`.

**Comment by KacperFKorban** [comment]
Ok, I think that I'll just display the type then.

File: scaladoc/src/dotty/tools/scaladoc/tasty/SyntheticSupport.scala [hunk 1]
```diff
@@ -75,11 +75,7 @@ object SyntheticsSupport:
     given ctx: dotc.core.Contexts.Context = quotes.asInstanceOf[scala.quoted.runtime.impl.QuotesImpl].ctx
     val sym = rsym.asInstanceOf[dotc.core.Symbols.Symbol]
     // `lookupPrefix` is private in `QuotesImpl#SymbolMethods`
-    val lookupPrefix =
-      if sym.isClass then
-        sym.thisType
-      else
-        sym.namedType
+    val lookupPrefix = sym.namedType
```
Comment (pikinier20) [inline scaladoc/src/dotty/tools/scaladoc/tasty/SyntheticSupport.scala:78]: This method is copied from `QuotesImpl` and I'm not sure if we should change it. Do you have any explanation there? Also, if we change the implementation there, we should remove the comment above

Comment (KacperFKorban) [reply scaladoc/src/dotty/tools/scaladoc/tasty/SyntheticSupport.scala:78]: The explanation is in `QuotesImpl`:
```
private def lookupPrefix: TypeRepr =
  if self.isClass then
    self.thisType // Needed to handle self-types (as in tests/run-macros/self)
  else
    self.namedType
```
Ok, will change the comment.

**Review by pikinier20 — APPROVED** [review]

**Pull request merged**